src/AdoNetCore.AseClient/Token/ClientCapabilityToken.cs

```diff
@@ -0,0 +1,462 @@
+using System;
+using System.IO;
+using AdoNetCore.AseClient.Enum;
+using AdoNetCore.AseClient.Interface;
+using AdoNetCore.AseClient.Internal;
+
+namespace AdoNetCore.AseClient.Token
+{
+    internal class ClientCapabilityToken : IToken
+    {
+        public TokenType Type => TokenType.TDS_CAPABILITY;
+
+        private readonly byte[] _capabilityBytes;
+
+        public ClientCapabilityToken(bool enableServerPacketSize = true)
+        {
+            var packetSizeRequestBit = enableServerPacketSize ? Byte6.REQ_SRVPKTSIZE : Byte6.None;
+
+            _capabilityBytes =  new[] {
+                //cap request
+                ((byte)Byte0.Unknown),
```

Comment:
Check out the structure of `TDS_CAPABILITY`

Byte0 and Byte16 are `TDS_CAP_REQUEST=1` and `TDS_CAP_RESPONSE=2` respectively. Following each type is a byte-length-prefixed array of capability bytes related to that type.

So in this case Byte1 and Byte17 each indicate there are 14 (0x0e) bytes following

Reply:
Derp.

Reply:
Have corrected this in the latest commit.